web/pdf_viewer.js

```diff
@@ -2186,14 +2190,26 @@ class PDFViewer {
     this.currentScaleValue = newScale;
   }
 
-  updateContainerHeightCss() {
-    const height = this.container.clientHeight;
-
-    if (height !== this.#previousContainerHeight) {
-      this.#previousContainerHeight = height;
+  #resizeObserverCallback(entries) {
+    for (const entry of entries) {
+      if (entry.target === this.container) {
+        const height = Math.floor(entry.borderBoxSize[0].blockSize);
+        if (height !== this.#previousContainerHeight) {
+          this.#previousContainerHeight = height;
+          docStyle.setProperty("--viewer-container-height", `${height}px`);
+        }
+      }
+    }
+  }
 
-      docStyle.setProperty("--viewer-container-height", `${height}px`);
+  get containerTopLeft() {
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("MOZCENTRAL")) {
+      return (this.#containerTopLeft ||= [
```

Comment:
Generally speaking, does caching the value actually do the right thing in all cases?

I noticed that `this.container.getBoundingClientRect()` will return *different values* depending on whether or not the sidebar is open.
This new method doesn't, for mozcentral-builds, take that into account and just caches whatever values that happens to be current at the first invocation; is this inconsistency not an issue here?

Reply:
*Possibly a dumb idea, but:* Could we perhaps use the caching (for all build targets), and simply clear it when the `ResizeObserver`-callback is invoked?

Reply:
Unfortunately it won't work.
When the sidebar is there and potentially resized, only `inset-inline-start` is changed but the overall container width is the same.
So I don't really see how to know when top/left are changed (especially left in the sidebar case).

Reply:
An idea could be to add a ResizeObserver on the sidebar container... but it isn't super exciting.

Reply:
I missed something... It works in fact.

Reply:
Thank you for the "dumb" idea :)